src/components/ec-modal/ec-modal.vue

```diff
@@ -0,0 +1,251 @@
+<template>
+  <transition name="ec-modal--fade">
+    <div
+      v-if="showModal"
+      class="ec-modal"
+      @click.self="closeModal()"
+    >
+      <div
+        class="ec-modal__content"
+        :class="{'ec-modal--lg': large}"
+      >
+        <header class="ec-modal__header">
+          <slot name="header" />
+          <ec-icon
+            v-if="isClosable"
+            class="ec-modal__close"
+            name="simple-close"
+            :size="24"
+            @click.stop="closeModal()"
+          />
+        </header>
+
+        <main class="ec-modal__main">
+          <slot name="main" />
+        </main>
+
+        <footer class="ec-modal__footer">
+          <div
+            v-if="showFooterLeftContent"
+            class="ec-modal__footer-left-content"
+          >
+            <slot name="footer-left-content" />
+          </div>
+
+          <button
+            v-if="this.$scopedSlots.negative"
+            class="ec-modal__negative-btn ec-btn ec-btn--md ec-btn--secondary ec-btn--rounded ec-modal__negative-btn--right"
```

Comment:
Pedantic: maybe group classes with the same block together? `ec-btn` classes first, `ec-modal__negative` second, or vice versa.

Reply:
I like this approach, too.